cmd/vicadmin/server.go

```diff
@@ -499,12 +507,41 @@ func (s *server) tailFiles(res http.ResponseWriter, req *http.Request, names []s
 	}
 }
 
+// deriveDisplayError takes a vSphere session error and returns
+// an error string that is safe to display on the vicadmin page.
+func deriveDisplayError(err error) string {
+	var displayErrStr string
+
+	if err != nil {
+		switch err := err.(type) {
+		case session.SDKURLError:
+			displayErrStr = fmt.Sprintf("SDK URL could not be parsed: %s", err.Err)
+		case session.SoapClientError:
+			displayErrStr = "unable to obtain a vim client"
+		case session.UserPassLoginError:
+			displayErrStr = "unable to log in with username and password"
+		default:
+			displayErrStr = err.Error()
+		}
+	}
+
+	return displayErrStr
+}
```

Comment:
I thought it would be cleaner to assign `displayErrStr` and return once at the end of the func, rather than putting several `return`s.